avm/src/anchor/main.rs

```diff
@@ -14,11 +14,15 @@ fn main() -> anyhow::Result<()> {
             version
         );
     }
-    Command::new(binary_path)
+    let exit = Command::new(binary_path)
         .args(args)
         .spawn()?
         .wait_with_output()
         .expect("Failed to run anchor-cli");
 
+    if !exit.status.success() {
```

Comment:
we might also just exit with `std::process::exit` anyways and not return a result because nothing else happen after this point.
not sure if this is regarded as clean or good practice so I will leave it like this. will change if desired of course.